src/dal/weblinklistener.ts

```diff
@@ -0,0 +1,101 @@
+import {inject, injectable} from "inversify";
+import {Logger} from "../bll/utils/logger";
+import {Disposable} from "vscode";
+import {HttpHostRequest} from "../bll/weblinklistener/httphostrequest";
+import {TYPES} from "../bll/utils/constants";
+import {Server as HttpServer, createServer as createHttpServer} from "http";
+import * as portscanner from "portscanner";
+import * as pify from "pify";
+
+@injectable()
+export class WebLinkListener implements Disposable {
+
+    private server: HttpServer;
+    private readonly sockets = {};
+    private static readonly SOCKET_NUMBER_START: number = 63330;
+    private static readonly SOCKET_NUMBER_END: number = WebLinkListener.SOCKET_NUMBER_START + 9;
+    public static readonly SUCCESS_STREAM: Uint8Array = new Uint8Array([
+            0x47, 0x49, 0x46, 0x38, 0x39, 0x61, 0x02, 0x00,
+            0x02, 0x00, 0x80, 0xFF, 0x00, 0xFF, 0xFF, 0xFF,
+            0x00, 0x00, 0x00, 0x2C, 0x00, 0x00, 0x00, 0x00,
+            0x01, 0x00, 0x01, 0x00, 0x00, 0x02, 0x02, 0x44,
+            0x01, 0x00, 0x3B
+        ]);
+
+    public static readonly FAILURE_STREAM: Uint8Array = new Uint8Array([
+            0x47, 0x49, 0x46, 0x38, 0x39, 0x61, 0x01, 0x00,
+            0x01, 0x00, 0x80, 0xFF, 0x00, 0xFF, 0xFF, 0xFF,
+            0x00, 0x00, 0x00, 0x2C, 0x00, 0x00, 0x00, 0x00,
+            0x01, 0x00, 0x01, 0x00, 0x00, 0x02, 0x02, 0x44,
+            0x01, 0x00, 0x3B
+        ]);
+
+    public constructor(@inject(TYPES.HttpHostRequest) private httpHostRequest: HttpHostRequest) {
+       this.initListener().then(() => {
+           Logger.logInfo(`WebLinkListener#initListener: listener was successfully attached to the port`);
+       }).catch((err) => {
+           Logger.logError(`WebLinkListener#initListener: ${err}`);
+        });
+    }
+
+    private async initListener(): Promise<void> {
+        this.server = createHttpServer((request, response) => {
+            this.httpHostRequest.processRequest(request).then((succeed) => {
+                WebLinkListener.writeResponse(response, succeed);
+            });
+        });
+
+        let nextSocketId: number = 0;
+        this.server.on("connection", (socket) => {
+            const socketId = nextSocketId++;
+            this.sockets[socketId] = socket;
```

Comment:
Why do we need to keep the list of sockets?

Reply:
I had some problems when after calling server.close() the port was not released.
So currently we collect sockets when they appear and remove them from collection when they are closed. When we want to release port we check whether open sockets exist and if so we close them.